go/cmd/zkctl/zkctl.go

```diff
@@ -63,20 +61,22 @@ func main() {
 	defer logutil.Flush()
 
 	fs := pflag.NewFlagSet("zkctl", pflag.ExitOnError)
+	fs.Usage = func() {
+		fmt.Fprintf(os.Stderr, usage)
+	}
+
 	log.RegisterFlags(fs)
 	logutil.RegisterFlags(fs)
-	_flag.Parse(fs)
-	args := _flag.Args()
+	args := servenv.ParseFlagsWithArgs("zkctl")
 
 	if len(args) == 0 {
-		flag.Usage()
+		fs.Usage()
```

Comment:
this doesn't include the global flags (defined on `servenv`'s global hooks), you would need to also grab a reference to the flag set that `servenv` is using (in an `OnParse` callback), store it in a variable, and then add those flags onto the `fs` variable you have here (via `fs.AddFlagSet`)

Reply:
I just realized that line#72 is unreachable because if len(args) is zero then  ParseFlagWithArgs will exit after printing message. Thus simplifying the logic.

https://github.com/vitessio/vitess/blob/main/go/vt/servenv/servenv.go#L344